subprojects/android-test/toast-rule/src/main/kotlin/com/avito/android/rule/internal/ProxyToastChecks.kt

```diff
@@ -0,0 +1,110 @@
+package com.avito.android.rule.internal
+
+import android.content.Context
+import android.view.View
+import android.widget.Toast
+import androidx.annotation.StringRes
+import androidx.test.espresso.matcher.ViewMatchers
+import com.avito.android.rule.ToastChecks
+import com.avito.android.test.waitFor
+import org.hamcrest.Description
+import org.hamcrest.Matcher
+import org.hamcrest.MatcherAssert
+import org.hamcrest.Matchers
+import org.hamcrest.Matchers.empty
+import org.hamcrest.Matchers.not
+import org.hamcrest.TypeSafeMatcher
+
+internal class ProxyToastChecks(
+    private val proxyToast: MockProxyToast,
+    private val textAccessor: ToastTextAccessor,
+) : ToastChecks {
+
+    private val longToastDuration = 3500L
+
+    override fun toastDisplayedWithText(text: String) {
+        internalToastDisplayedWithText(text)
+    }
+
+    override fun toastDisplayedWithText(@StringRes resId: Int) {
+        val toast = firstToastOrThrow()
+        val context = toast.getFieldByReflection("mContext") as Context
```

Comment:
Are you sure we should use reflection here?
Could we simply use context from `Instrumentation`?

Reply:
Thanks, didn't think of it. 
Fixed

To be correct, application context is not a replacement. There is a case of configuration-specific resources. 
But it looks like a corner case. We can live without it.